Tina/content Various text and content changes.

QueenChristina opened the pull request:
Text and content changes according to QA - Tool Content doc (all except info popup).

Additionally, the textField and Grocery list instructions now have matching font size according to QA - Visuals doc.

Also, updated utils.js "pluralize" function logic to allow plural term to be used with values below 1 (eg. "0.8 liters" instead of "0.8 liter".

@jeffkassab Feel free to double check content + suggest further changes if need be.

Issues

- [Code quality] Right now, App.js uses a lot of "magic numbers" when calculating amounts of things and converting between units eg. totalWaterUse, etc.. I suggest keeping a set of predefined constants somewhere so unit conversion is easier and more intuitive.

QueenChristina committed:
Text content changes to titles and added subtitles

QueenChristina committed:
Merge branch 'dev' into tina/content

QueenChristina committed:
Text content change in Comparison slider labels and summary

QueenChristina committed:
Changed font size style of GroceryList instructions to match textField input's

QueenChristina committed:
Various content changes: changed units and content in Summary, minor text content change in subtitles and labels Comparison, Grocery instrustions, and BarChart

QueenChristina committed:
Update logic of pluralize to allow decimal values below 1 to use plural term rather than singular

QueenChristina committed:
Merge branch 'dev' into tina/content

tonychen2 commented on frontend/src/utils.js:
```diff
@@ -8,11 +8,10 @@ export function pluralize(singular, plural, count) {
-    if (count > 1) {
-        return plural
-    } else {
+    if (count == 1) {
```
Are you sure about this? My research showed it's "0.5 inch" rather than "0.5 inches"?

Also, thing to remember: float equality is dangerous because 0.0 === 0.0 is not guaranteed because the numbers are not necessarily exact. That's why people usually replace equality evaluation with checking that the difference is smaller than a tiny threshold (e.g. 1e-5). If you are interested you can look into how floating point representation works.

QueenChristina replied:
Not sure since I didn’t do any research. But “0.5 inches” (“zero point five”) sounded more natural to me than “0.5 inch” unless you’re saying it like “half of an inch” or “0.5 of an inch”. Maybe we can ask at standup tomorrow what the others think. Doing research now though and I see most forums seem to think “0.5 inches” is more popular possibly because it implies “five tenths of a mile” and the “five tenths” is quite plural. 

Thank you for the reminder about floating equality. I had forgotten about it.

tonychen2 commented on frontend/src/App.js:
```diff
@@ -182,8 +182,11 @@ class App extends Component {
-                totalWaterUse: totalWaterUse * 4.2
+                // parkingEq: totalLandUse / 14, // Convert from sq meters to # parking spots
+                totalTreeUse: totalLandUse / 256, // 256sqft/tree Convert from land use to Californian tree use
+                totalWaterUse: totalWaterUse * 4.2.toExponential(), // Water use in cups
```
I don't think exponential representation is very intuitive to people.

QueenChristina replied:
Thanks for catching this error. Honestly not sure how this got there, maybe it snuck in from some of VScode’s autocompletion? I’ll be more careful from now on.

QueenChristina committed:
Fix floating point equality logic of pluralize, fix summary plural text content and formulas

tonychen2 committed:
Update bar chart instructions for recommendation

Food names are not currently clickable so we make it clear that the bars are to be clicked

tonychen2 committed:
Merge branch 'dev' into tina/content
